docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,707 @@
+# ML.NET Cookbook
+
+This document is intended to provide essential samples for common usage patterns of ML.NET. 
+It is advisable to be at least minimally familiar with [high-level concepts of ML.NET](MlNetHighLevelConcepts.md), otherwise the terminology in this document may be foreign to you.
+
+## How to use this cookbook
+
+Developers often work by copying and pasting source code from somewhere and then adapting it to their needs. We do it all the time.
+
+So, we decided to embrace the pattern and provide an authoritative set of example usages of ML.NET, for many common scenarios that you may encounter.
+These examples are multi-purpose:
+
+- They can kickstart your development, so that you don't start from nothing,
+- They are annotated and verbose, so you have easier time adapting them to your needs.
+
+Each sample also contains a snippet of the data file used in the sample. We mostly use snippets from our test datasets for that.
+
+Please feel free to search this page and use any code that suits your needs.
+
+### List of recipes
+
+- [How do I load data from a text file?](#how-do-i-load-data-from-a-text-file)
+- [How do I load data with many columns from a CSV?](#how-do-i-load-data-with-many-columns-from-a-csv)
+- [How do I look at the intermediate data?](#how-do-i-look-at-the-intermediate-data)
+- [How do I train a regression model?](#how-do-i-train-a-regression-model)
+- [How do I verify the model quality?](#how-do-i-verify-the-model-quality)
+- [How do I save and load the model?](#how-do-i-save-and-load-the-model)
+- [How do I use the model to make one prediction?](#how-do-i-use-the-model-to-make-one-prediction)
+- [What if my training data is not in a text file?](#what-if-my-training-data-is-not-in-a-text-file)
+- [I want to look at my model's coefficients](#i-want-to-look-at-my-models-coefficients)
+- [What is normalization and why do I need to care?](#what-is-normalization-and-why-do-i-need-to-care)
+- [How do I train my model on categorical data?](#how-do-i-train-my-model-on-categorical-data)
+- [How do I train my model on textual data?](#how-do-i-train-my-model-on-textual-data)
+- [How do I train using cross-validation?](#how-do-i-train-using-cross-validation)
+
+## How do I load data from a text file?
+
+`TextLoader` is used to load data from text files. You will need to specify what are the data columns, what are their types, and where to find them in the text file. 
+
+Note that it's perfectly acceptable to read only some columns of a file, or read the same column multiple times.
+
+Example file (https://github.com/dotnet/machinelearning/blob/master/test/data/adult.tiny.with-schema.txt):
+```
+Label	Workclass	education	marital-status
+0	Private	11th	Never-married
+0	Private	HS-grad	Married-civ-spouse
+1	Local-gov	Assoc-acdm	Married-civ-spouse
+1	Private	Some-college	Married-civ-spouse
+
+```
+
+This is how you can read this data:
+```c#
```

Comment:
>c# [](start = 3, length = 2)

Just a minor note, while I think github markdown supports the `c#` token to indicate C#, most other markdown parsers I've encountered (including Visual Studio Code, which I tend to use, and the thing I used for the web help application) do not. It might be nice if you have time to just change these to the more universally recognized `csharp`.